core/src/main/java/org/pentaho/platform/util/Emailer.java

```diff
@@ -261,89 +289,105 @@ public boolean send() {
 
         if ( body != null ) {
           //We need to add message to the top of the email body
-          final MimeMultipart oldMultipart = (MimeMultipart) msg.getContent();
-          final MimeMultipart newMultipart = new MimeMultipart( "related" );
-
-
-          for ( int i = 0; i < oldMultipart.getCount(); i++ ) {
-            BodyPart bodyPart = oldMultipart.getBodyPart( i );
-
-            final Object content = bodyPart.getContent();
-            //Main HTML body
-            if ( content instanceof String ) {
-              final String newContent = body + "<br/><br/>" + content;
-              final MimeBodyPart part = new MimeBodyPart();
-              part.setText( newContent, "UTF-8", "html" );
-              newMultipart.addBodyPart( part );
-            } else {
-              //CID attachments
-              newMultipart.addBodyPart( bodyPart );
-            }
-          }
-
+          setBodyForEmbeddedHtml( msg, body );
 
-          msg.setContent( newMultipart );
         }
       } else {
 
         // construct the message
         msg = new MimeMessage( session );
-        Multipart multipart = new MimeMultipart();
-
-        if ( body != null ) {
-          MimeBodyPart bodyMessagePart = new MimeBodyPart();
-          bodyMessagePart.setText( body, LocaleHelper.getSystemEncoding() );
-          multipart.addBodyPart( bodyMessagePart );
-        }
-
-        if ( attachment != null ) {
-          ByteArrayDataSource dataSource = new ByteArrayDataSource( attachment, attachmentMimeType );
-          // attach the file to the message
-          MimeBodyPart attachmentBodyPart = new MimeBodyPart();
-          attachmentBodyPart.setDataHandler( new DataHandler( dataSource ) );
-          attachmentBodyPart.setFileName( attachmentName );
-          multipart.addBodyPart( attachmentBodyPart );
-        }
+        Multipart multipart = setMultipart( body );
```

Comment:
Good change on moving this logic to adding attachment into wrapper functions `setMultipart` and `setBodyForEmbeddedHtml` . The original way, made it harder to read the code, the nuance details rightfully so should be hidden.